tests/client/asynchronous_test.py

```diff
@@ -49,3 +54,76 @@ def test_matches(self):
     def assert_response_type(self, matcher, path, expected_response_type):
         response = json.loads(matcher.response(path))
         assert response["response-type"] == expected_response_type
+
+
+class TestResolveHost:
+    @pytest.mark.asyncio
+    @pytest.mark.parametrize(
+        "num_of_dns_resp, num_clients",
+        [(1, 256), (2, 128), (3, 1), (3, 64), (4, 16), (3, 2000)],
+    )
+    @mock.patch("esrally.client.asynchronous.aiohttp.ThreadedResolver.resolve")
+    async def test_resolve_host_even_client_allocation(
+        self,
+        mocked_resolver,
+        num_of_dns_resp,
+        num_clients,
+    ):
+        """
+        TCPConnector._resolve_host() returns all the IPs a given name resolves to, but the underlying connection but the
+        logic in TCPConnector._create_direct_connection() only ever selects the first succesful host from this list.
+
+        So, we use the factory assigned client_id to deterministically return an IP and then place it at the beginning
+        of the list to evenly distributes connections across _all_ clients.
+
+        https://github.com/elastic/rally/issues/1598
+        """
```

Comment:
We already said that above, I don't think repeating it helps. I think I would just remove those lines?

Reply:
👍 addressed in https://github.com/elastic/rally/pull/1707/commits/fe762dfdc66ad80bec07fbcb3a0d4be7eec11a64